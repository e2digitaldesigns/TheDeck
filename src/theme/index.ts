export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
