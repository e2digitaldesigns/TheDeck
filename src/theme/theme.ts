export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
