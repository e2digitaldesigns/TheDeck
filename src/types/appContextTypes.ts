export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
