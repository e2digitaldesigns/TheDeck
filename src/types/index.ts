/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
