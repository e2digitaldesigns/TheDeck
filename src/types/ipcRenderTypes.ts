export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
