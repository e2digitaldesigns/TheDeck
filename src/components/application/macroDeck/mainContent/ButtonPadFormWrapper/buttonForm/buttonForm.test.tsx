import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import _cloneDeep from "lodash/cloneDeep";
import ButtonForm from "./buttonForm";

import { ThemeProvider } from "styled-components";
import { theme } from "./../../../../../../theme/theme";

import initAppState from "../../../../../../__mock_data__/appState.json";
import initGlobalState from "../../../../../../__mock_data__/globalState.json";

import { IntAppData, IntGlobalData } from "../../../../../../types";

const mockAppState: IntAppData = {
  appState: _cloneDeep(initAppState),
  setAppState: jest.fn(appState => appState)
};

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

const mockHooks = {
  useAppData: mockAppState,
  useGlobalData: mockGlobalState,
  getActiveButton: () => mockGlobalState.state.buttonPads[0],
  getActiveButtonIndex: () => 0,
  updateButtonPad: jest.fn()
};

jest.mock("../../../../../../hooks", () => ({
  useAppData: () => mockHooks.useAppData,

  useButton: () => ({
    getActiveButton: mockHooks.getActiveButton,
    getActiveButtonIndex: mockHooks.getActiveButtonIndex,
    updateButtonPad: mockHooks.updateButtonPad
  }),

  useGlobalData: () => mockGlobalState
}));

const testSetup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ButtonForm />
    </ThemeProvider>
  );
};

beforeEach(() => {
  mockGlobalState.state = _cloneDeep(initGlobalState);
  mockAppState.appState = _cloneDeep(initAppState);
});

afterEach(() => {
  jest.clearAllMocks();
});

describe("<ButtonForm/>", () => {
  it("Should render without errors", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should enter text in text field", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    const textField = wrapper.getByTestId(
      "button_form__text-field-text"
    ) as HTMLInputElement;
    expect(textField).toBeTruthy();

    const value = "Hello";
    fireEvent.change(textField, { target: { value } });
    expect(textField.value).toBe(value);
  });

  it("Should enter text validation", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    const textField = wrapper.getByTestId(
      "button_form__text-field-text"
    ) as HTMLInputElement;
    expect(textField).toBeTruthy();

    const value = "H!!!ello Hell^%^o He@l#";
    const check = "Hello Hello Hel";
    fireEvent.change(textField, { target: { value } });
    expect(textField.value).toBe(check);
  });

  it("Should change text color", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    const textColor = wrapper.getByTestId(
      "button_form__text-field-text-color"
    ) as HTMLInputElement;
    expect(textColor).toBeTruthy();

    const value = "#111111";
    fireEvent.change(textColor, { target: { value } });
    expect(textColor.value).toBe(value);
  });

  it("Should change bg color", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    const bgColor = wrapper.getByTestId(
      "button_form__bg-color"
    ) as HTMLInputElement;
    expect(bgColor).toBeTruthy();

    const value = "#111111";
    fireEvent.change(bgColor, { target: { value } });
    expect(bgColor.value).toBe(value);
  });

  it("Should use default bgColor", () => {
    const wrapper = testSetup();
    const bgColor = wrapper.getByTestId(
      "button_form__bg-color"
    ) as HTMLInputElement;
    expect(bgColor.value).toBe("#555555");
  });

  it("Should use button bgColor", () => {
    mockGlobalState.state.buttonPads[0].bgColor = "#ffffff";
    const wrapper = testSetup();
    const bgColor = wrapper.getByTestId(
      "button_form__bg-color"
    ) as HTMLInputElement;
    expect(bgColor.value).toBe("#ffffff");
  });

  it("Should have disabled buttons", () => {
    const wrapper = testSetup();

    const submitButton = wrapper.getByTestId("button_form__submit");
    expect(submitButton).toBeTruthy();
    expect(submitButton).toHaveAttribute("disabled");

    const resetButton = wrapper.getByTestId("button_form__reset");
    expect(resetButton).toBeTruthy();
    expect(resetButton).toHaveAttribute("disabled");

    const revertButton = wrapper.getByTestId("button_form__revert");
    expect(revertButton).toBeTruthy();
    expect(revertButton).toHaveAttribute("disabled");
  });

  it("Should have enabled buttons", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();

    const submitButton = wrapper.getByTestId("button_form__reset");
    expect(submitButton).toBeTruthy();
    expect(submitButton).not.toHaveAttribute("disabled");
  });

  it("Should submit form", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    const textField = wrapper.getByTestId("button_form__text-field-text");
    const submit = wrapper.getByTestId("button_form__submit");

    fireEvent.change(textField, { target: { value: "go go go" } });
    fireEvent.click(submit);
    expect(mockHooks.updateButtonPad).toHaveBeenCalledTimes(1);
  });

  it("Should revert to saved", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    const textField = wrapper.getByTestId(
      "button_form__text-field-text"
    ) as HTMLInputElement;
    const revertButton = wrapper.getByTestId("button_form__revert");
    fireEvent.change(textField, { target: { value: "go go go" } });
    fireEvent.click(revertButton);
    expect(textField.value).toBe(mockGlobalState.state.buttonPads[0].text);
  });

  it("Should reset default", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;
    const wrapper = testSetup();
    const textField = wrapper.getByTestId(
      "button_form__text-field-text"
    ) as HTMLInputElement;
    const resetButton = wrapper.getByTestId("button_form__reset");
    fireEvent.change(textField, { target: { value: "going going gone" } });
    fireEvent.click(resetButton);
    expect(textField.value).toBe(mockGlobalState.state.buttonPads[0].text);
  });
});
