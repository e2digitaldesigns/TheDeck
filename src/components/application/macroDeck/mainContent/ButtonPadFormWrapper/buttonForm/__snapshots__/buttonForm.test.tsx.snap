// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonForm/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-jRQBWg jyCeeA"
      >
        <fieldset
          class="sc-bdvvtL sc-gKclnd Smgns fzBRvr"
        >
          <div>
            <label
              for="text"
            >
              Text:
            </label>
            <input
              class="sc-gsDKAQ sc-iCfMLu dEgDrl kUuFGE"
              data-testid="button_form__text-field-text"
              name="text"
              type="text"
              value="Working"
            />
            <input
              data-testid="button_form__text-field-text-color"
              name="textColor"
              type="color"
              value="#cccccc"
            />
          </div>
          <div>
            <label
              for="icon"
            >
              Icon:
            </label>
            <input
              class="sc-gsDKAQ sc-iCfMLu dEgDrl kUuFGE"
              data-testid="button_form__text-field-icon"
              readonly=""
              type="text"
              value="Image"
            />
            <input
              data-testid="button_form__icon-color"
              name="iconColor"
              type="color"
              value="#cccccc"
            />
          </div>
          <div>
            <label
              for="bgColor"
            >
              BG Color:
            </label>
            <input
              class="sc-pVTFL vgiIg"
              data-testid="button_form__bg-color"
              name="bgColor"
              type="color"
              value="#555555"
            />
          </div>
        </fieldset>
        <div>
          <button
            class="sc-hKwDye sc-jrQzAO PLLBx dgZkmA"
            data-testid="button_form__submit"
            disabled=""
          >
            Save
          </button>
          <button
            class="sc-hKwDye sc-jrQzAO PLLBx dgZkmA"
            data-testid="button_form__revert"
            disabled=""
          >
            Revert to Saved
          </button>
          <button
            class="sc-hKwDye sc-jrQzAO PLLBx dgZkmA"
            data-testid="button_form__reset"
          >
            Reset to Default
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-jRQBWg jyCeeA"
    >
      <fieldset
        class="sc-bdvvtL sc-gKclnd Smgns fzBRvr"
      >
        <div>
          <label
            for="text"
          >
            Text:
          </label>
          <input
            class="sc-gsDKAQ sc-iCfMLu dEgDrl kUuFGE"
            data-testid="button_form__text-field-text"
            name="text"
            type="text"
            value="Working"
          />
          <input
            data-testid="button_form__text-field-text-color"
            name="textColor"
            type="color"
            value="#cccccc"
          />
        </div>
        <div>
          <label
            for="icon"
          >
            Icon:
          </label>
          <input
            class="sc-gsDKAQ sc-iCfMLu dEgDrl kUuFGE"
            data-testid="button_form__text-field-icon"
            readonly=""
            type="text"
            value="Image"
          />
          <input
            data-testid="button_form__icon-color"
            name="iconColor"
            type="color"
            value="#cccccc"
          />
        </div>
        <div>
          <label
            for="bgColor"
          >
            BG Color:
          </label>
          <input
            class="sc-pVTFL vgiIg"
            data-testid="button_form__bg-color"
            name="bgColor"
            type="color"
            value="#555555"
          />
        </div>
      </fieldset>
      <div>
        <button
          class="sc-hKwDye sc-jrQzAO PLLBx dgZkmA"
          data-testid="button_form__submit"
          disabled=""
        >
          Save
        </button>
        <button
          class="sc-hKwDye sc-jrQzAO PLLBx dgZkmA"
          data-testid="button_form__revert"
          disabled=""
        >
          Revert to Saved
        </button>
        <button
          class="sc-hKwDye sc-jrQzAO PLLBx dgZkmA"
          data-testid="button_form__reset"
        >
          Reset to Default
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
